randovania/generator/dock_weakness_distributor.py

```diff
@@ -279,7 +279,9 @@ def _determine_valid_weaknesses(
 
     if state is not None:
         reach = ResolverReach.calculate_reach(logic, state)
-        if state.node == target:
+        if state.node == target or isinstance(state.node, DockLockType) and state.node.dock == target:
```

Comment:
is it really possible that state.node can be either a docklocktype or a docknode?

Reply:
my immediate intuition is that it would certainly be possible. if anything, it's more likely to be a docklocknode than a docknode I think

Reply:
I'm convinced that `DockLockType` was a slip, but I found through experience that they can be `DockLockNode`.